### PR TITLE
test(pagination): add common test suite

### PR DESCRIPTION
**Pull request by sarahdayan**
This reworks the tests for the `pagination` widget to move them in the common test suite and reduce flavor-specific tests.

https://algolia.atlassian.net/browse/FX-2516

### CODE REVIEW

File: packages/vue-instantsearch/src/components/Pagination.vue [hunk 1]
```diff
@@ -1,5 +1,8 @@
 <template>
-  <div v-if="state" :class="suit()">
+  <div
+    v-if="state"
+    :class="{ [suit()]: true, [suit('', 'noRefinement')]: state.nbPages <= 1 }"
```
Comment (sarahdayan) [inline packages/vue-instantsearch/src/components/Pagination.vue:4]: This is additive, so it shouldn't be considered breaking IMO.

Worst case, if users have snapshot tests they can break, but this won't break production.

File: tests/utils/normalizeSnapshot.ts [hunk 2]
```diff
@@ -16,8 +16,8 @@ export function normalizeSnapshot(html: string) {
       // Vue renders extra whitespace between span elements
       .replace(/<\/span> <span/g, '</span><span')
       // Vue renders an empty comment for falsy v-if predicates
-      .replace(/(\s+)?<!---->/g, '') // Vue 2
-      .replace(/(\s+)?<!--v-if-->/g, '') // Vue 3
+      .replace(/(\s+)?<!---->(\s+)?/g, '') // Vue 2
+      .replace(/(\s+)?<!--v-if-->(\s+)?/g, '') // Vue 3
```
Comment (sarahdayan) [inline tests/utils/normalizeSnapshot.ts:20]: Vue also adds space _after_ :table-flip:

File: tests/common/widgets/pagination/options.ts
[the remark this answers is not in the document]
Comment (sarahdayan) [reply tests/common/widgets/pagination/options.ts:1]: Yes, we went a little overboard with snapshots in this one when we did the RIS test suite. I'll take this PR's opportunity to clean them up a bit.

[the remark this answers is not in the document]
Comment (sarahdayan) [reply tests/common/widgets/pagination/options.ts:1]: Addressed in https://github.com/algolia/instantsearch/pull/5836/commits/5e0d4bb5f72f4d5caa5f221bc53175a989fa8a68.

**Review by dhayab — APPROVED** [review]

File: packages/instantsearch.js/src/components/Pagination/Pagination.tsx [hunk 3]
```diff
@@ -158,8 +158,8 @@ function PaginationLink({
     <li
       className={cx(
         cssClasses.item,
-        className,
```
[the remark this answers is not in the document]
Comment (sarahdayan) [reply packages/instantsearch.js/src/components/Pagination/Pagination.tsx:161]: Oops I forgot about them, on it.

[the remark this answers is not in the document]
Comment (sarahdayan) [reply packages/instantsearch.js/src/components/Pagination/Pagination.tsx:161]: Addressed in [28ae431](https://github.com/algolia/instantsearch/pull/5836/commits/28ae4315a0a343ed20946652201f0858b0be79c6).

**Review by Haroenv — APPROVED** [review]

**Pull request merged**